### PR TITLE
Implementation of instant-apps-language-x components for app translation user experience.

### PR DESCRIPTION
**Pull request by rslibed**
Implementation of instant-apps-language-x components for app translation workflow.

Components:

- instant-apps-language-translator
    - instant-apps-language-translator-item
    - instant-apps-language-translator-search
- instant-apps-language-switcher
- instant-apps-ckeditor-wrapper

### CODE REVIEW

File: packages/instant-apps-components/src/assets/t9n/instant-apps-language-translator/resources.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    "sk": "Slovak",
+    "sl": "Slovenian",
+    "sr": "Serbian",
+    "sv": "Swedish",
+    "th": "Thai",
+    "tr": "Turkish",
+    "uk": "Ukrainian",
+    "vi": "Vietnamese",
+    "zh-CN": "Simplified Chinese",
+    "zh-HK": "Traditional Chinese",
```
Comment (kellyhutchins) [inline packages/instant-apps-components/src/assets/t9n/instant-apps-language-translator/resources.json:47]: We might want to add Hong Kong and Taiwan after traditional Chinese for zh-CN and zh-HK.

File: packages/instant-apps-components/src/components/instant-apps-language-switcher/instant-apps-language-switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        eventData['data'] = this.t9nData[translatedLanguage];
+      }
+      this.intl.setLocale(this.selectedLanguage);
+
+      // Set url parameter 'locale' with value
+      const params = new URLSearchParams(window.location.search);
+      params.set('locale', this.selectedLanguage);
+      document.documentElement.setAttribute('lang', this.selectedLanguage);
+
+      if (this.selectedLanguage === 'ar' || this.selectedLanguage === 'he') {
```
Comment (kellyhutchins) [inline packages/instant-apps-components/src/components/instant-apps-language-switcher/instant-apps-language-switcher.tsx:133]: This is fine if you aren't loading the api but if you have to do it for this project might be worth using the esri/intl prefersRTL instead? https://developers.arcgis.com/javascript/latest/api-reference/esri-intl.html#prefersRTL

File: packages/instant-apps-components/src/instant-apps-language-translator.html [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+ */ -->
+
+<!DOCTYPE html>
+<html dir="ltr">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
+    <title>ArcGIS Instant Apps Components</title>
+
+    <link rel="stylesheet" href="https://js.arcgis.com/4.27/@arcgis/core/assets/esri/themes/light/main.css" />
```
Comment (kellyhutchins) [inline packages/instant-apps-components/src/instant-apps-language-translator.html:16]: Update these references to 4.28 at some point

File: packages/instant-apps-components/src/instant-apps-language-translator.html [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+              <calcite-button id="openLangTranslator">Open language translator</calcite-button>
+            </calcite-label>
+          </div>
+        </calcite-panel>
+      </calcite-shell-panel>
+      <div id="viewDiv"></div>
+    </calcite-shell>
+    <instant-apps-language-translator tabindex="-1" id="langTranslator"></instant-apps-language-translator>
+    <script type="module">
+      import MapView from 'https://js.arcgis.com/4.27/@arcgis/core/views/MapView.js';
```
Comment (kellyhutchins) [inline packages/instant-apps-components/src/instant-apps-language-translator.html:56]: Same as above update to 4.28...doesn't have to be now.

**Review by kellyhutchins — APPROVED** [review]
Looks good Ryan

File: packages/instant-apps-components/src/components/instant-apps-language-switcher/instant-apps-language-switcher.tsx (hunk 2 above)
Comment (rslibed) [reply packages/instant-apps-components/src/components/instant-apps-language-switcher/instant-apps-language-switcher.tsx]: Done

File: packages/instant-apps-components/src/assets/t9n/instant-apps-language-translator/resources.json (hunk 1 above)
Comment (rslibed) [reply packages/instant-apps-components/src/assets/t9n/instant-apps-language-translator/resources.json:47]: Done

**Pull request merged**